text/0000-extended-standard-library.md

```diff
@@ -0,0 +1,370 @@
+
+2. Create an ESL GitHub Organization
+
+A GitHub organization should be created for the ESL. It should have its own GitHub organization to make it clear to Rust developers what is part of the ESL.
+
+The ESL GitHub organization should host ESL crates as well as policy, governance, and ESL-specific testing and infrastructure repositories.
+
+The ESL GitHub organization will be administered by the Rust Libraries ESL sub-team.
+
+3. Create Initial Policies
+
+The ESL community will decide on an initial set of policies, possibly similar in scope to the examples in the RFC.
+
+ESL policies will be published in a policy repository in the ESL GitHub organization.
+
+The ESL GitHub organization should be configured and administered by the Rust Libraries ESL sub-team in line with published ESL policies.
+
+4. Create an Administrative and Infrastructure Support Relationship
+
+The ESL community will engage the Rust Foundation to assist with administering finances, hiring any necessary contractors and/or staff, and operating supporting infrastructure. This may involve becoming an official Rust Foundation project.
+
+Regardless of the details of the relationship to the Rust Foundation, the Rust Foundation will play a supporting role, much like it does for the Rust language, rather than making decisions for the ESL and its maintainers.
+
+5. Secure Initial Funding Commitments
+
+Work on the ESL will require resources to set up the necessary structure and assist maintainers with meeting ESL requirements and general maintenance. Additional resources should also be made available to the Rust Libraries team as necessary to assist with performance of their role.
+
+The [Rust Foundation](https://rustfoundation.org/) and [ISRG](https://www.abetterinternet.org/) are willing to assist with raising the necessary funds.
+
+6. Create the ESL
+
+At this point the initial ESL crates will move into the ESL GitHub organization.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+1) This will require making some difficult decisions.
+
+This project will require being able to make difficult decisions, such as selecting functionality to include and deciding on policies. ESL community and leadership should recognize this and work in such a way that difficult decisions can be made when necessary.
+
+2) Managing community and maintainer relationships may be challenging.
+
+Maintainers (and perhaps community members, generally), could be upset about the idea of “picking winners” from the ecosystem. If such a situation arises, the ESL community should strive to reach consensus among the people involved, while at the same time keeping analysis paralysis at bay. At the end of the day, making a decision is important even if it is not "perfect" by a given set of standards. In any case, the ESL providing specific functionality does not stop others from offering alternatives.
```

Comment:
The drawback of picking winners isn't that folks "could be upset", it's the (good) reasons for which they would be made upset. If you maintain a crate whose competitor is selected for inclusion in the ESL, there will begin to be tremendous pressure to avoid your crate. If you have an idea for an alternative approach to a problem "solved" by the ESL, there will be tremendous pressure to avoid your crate.

Had the ESL existed in 2018, I don't think I'd be paying my bills by maintaining one of the crates I maintain today; there simply would have been too much pressure to use the then-dominant alternative.

Reply:
I somewhat feel like this is already true today. I think the fundamental difference between the status quo and this proposal is that this proposal has a (presumably small) group of people making a determination about the winner as compared today where a winner tends to manifest via inertia or other pressures. But we certainly don't need an ESL to have the problem of it being almost impossible to unseat an existing dominant solution.

Reply:
But I think we can definitely avoid making it worse at the least

Reply:
There's also the issue that, in many cases, there may be multiple crates for a given purpose which make very different design decisions. Often, these design decisions are equally "correct", and the right choice depends on the particular use case. In a situation like this where two or more crates that have a comparable level of maintenance quality make substantially different and incompatible design choices, what would the ESL do? Would both alternatives be added to the ESL? Or would the ESL just choose one such crate and leave users whose particular use case prefers the other design without an ESL-blessed option?

Reply:
Also worth calling out that choosing only one competitor means that the other ESL crates are forced to make the same pick. Basically...

> We'd love for your crate to be part of ESL, but of course you'd have to switch from `fastrand` to `rand`. I'm sure your users won't mind!